code/modules/atmospherics/machinery/components/electrolyzer/electrolyzer.dm

```diff
@@ -69,41 +69,60 @@
 	if(!on)
 		return PROCESS_KILL
 
-	if(!cell || cell.charge <= 0)
+	if((!cell || cell.charge <= 0) && !anchored)
 		on = FALSE
 		update_appearance()
 		return PROCESS_KILL
 
-	var/turf/L = loc
-	if(!istype(L))
+	var/turf/our_turf = loc
+	if(!istype(our_turf))
 		if(mode != ELECTROLYZER_MODE_STANDBY)
 			mode = ELECTROLYZER_MODE_STANDBY
 			update_appearance()
 		return
 
-	var/newMode = on ? ELECTROLYZER_MODE_WORKING : ELECTROLYZER_MODE_STANDBY //change the mode to working if the machine is on
+	var/new_mode = on ? ELECTROLYZER_MODE_WORKING : ELECTROLYZER_MODE_STANDBY //change the mode to working if the machine is on
 
-	if(mode != newMode) //check if the mode is set correctly
-		mode = newMode
+	if(mode != new_mode) //check if the mode is set correctly
+		mode = new_mode
 		update_appearance()
 
 	if(mode == ELECTROLYZER_MODE_STANDBY)
 		return
 
-	var/datum/gas_mixture/env = L.return_air() //get air from the turf
-	var/datum/gas_mixture/removed = env.remove_ratio(0.1)
+	var/datum/gas_mixture/env = our_turf.return_air() //get air from the turf
 
-	if(!removed)
+	if(!env)
 		return
 
-	removed.assert_gases(/datum/gas/water_vapor, /datum/gas/oxygen, /datum/gas/hydrogen)
-	var/proportion = min(removed.gases[/datum/gas/water_vapor][MOLES], (1.5 * delta_time * workingPower))//Works to max 12 moles at a time.
-	removed.gases[/datum/gas/water_vapor][MOLES] -= proportion * 2 * workingPower
-	removed.gases[/datum/gas/oxygen][MOLES] += proportion * workingPower
-	removed.gases[/datum/gas/hydrogen][MOLES] += proportion * 2 * workingPower
-	env.merge(removed) //put back the new gases in the turf
+	call_reactions(env, delta_time)
+
 	air_update_turf(FALSE, FALSE)
-	cell.use((5 * proportion * workingPower) / (efficiency + workingPower))
+
+	if(anchored)
+		return
+
+	cell.use((5 * (1.5 * delta_time * working_power) * working_power) / (efficiency + working_power))
+
+/obj/machinery/electrolyzer/proc/call_reactions(datum/gas_mixture/environment, delta_time)
+	for(var/reaction in GLOB.electrolyzer_reactions)
+		var/datum/electrolyzer_reaction/current_reaction = GLOB.electrolyzer_reactions[reaction]
+		var/list/reqs = current_reaction.requirements
+		if(!reaction_check(reqs, environment))
+			continue
+		current_reaction.react(loc, working_power, delta_time)
```

Comment:
oh mmm. seems a bit silly to pull out the gasmix twice and pass it around like this, why are ya doing that?

Reply:
true, i've removed the extra calls